src/lib.rs

```diff
@@ -30,62 +32,87 @@ pub struct Hx711<IN, OUT> {
     mode: Mode,
 }
 
-impl<IN, OUT> Hx711<IN, OUT>
+
+impl<IN, OUT, PINERR> Hx711<IN, OUT>
 where
-    IN: InputPin,
-    OUT: OutputPin,
+    IN: InputPin<Error = PINERR>,
+    OUT: OutputPin<Error = PINERR>,
 {
     /// Creates a new driver from Input and Outut pins
-    pub fn new(dout: IN, mut pd_sck: OUT) -> Self {
-        pd_sck.set_low();
-        let mut hx711 = Hx711 {
+    pub fn new(dout: IN, pd_sck: OUT) -> Self {
+        let hx711 = Hx711 {
             dout,
             pd_sck,
             mode: Mode::ChAGain128,
         };
-        hx711.reset();
         hx711
     }
 
-    /// Set the mode (channel and gain).
-    pub fn set_mode(&mut self, mode: Mode) {
-        self.mode = mode;
-        block!(self.retrieve()).unwrap();
+    /// Destruct the device and hand back the pins
+    pub fn destroy(self) -> (IN, OUT) {
+        (self.dout, self.pd_sck)
     }
 
-    /// Reset the chip. Mode is Channel A Gain 128 after reset.
-    pub fn reset(&mut self) {
-        self.pd_sck.set_high();
-        for _ in 1..3 {
-            self.dout.is_high();
-        }
-        self.pd_sck.set_low();
+    /// Enable 
+    pub fn enable(&mut self) -> Result<(), PINERR> {
+        self.pd_sck.set_low()
+    }
+
+    /// Disable - after 60 us of clk high the chip will go to sleep
+    pub fn disable(&mut self) -> Result<(), PINERR> {
+        self.pd_sck.set_high()
+    }
+
+    /// Set the mode (channel and gain).
+    pub fn set_mode<DELAY>(&mut self, mode: Mode, delay: &mut DELAY) -> Result<(), PINERR> 
+    where
+        DELAY: DelayUs<u16>,
+    {
+        self.mode = mode;
+        block!(self.retrieve(delay)).map(|_| ())
     }
 
     /// Retrieve the latest conversion value if available
-    pub fn retrieve(&mut self) -> nb::Result<i32, !> {
-        self.pd_sck.set_low();
-        if self.dout.is_high() {
+    pub fn retrieve<DELAY>(&mut self, delay: &mut DELAY) -> nb::Result<i32, PINERR>
+    where
+        DELAY: DelayUs<u16>,
+    {
+        self.pd_sck.set_low()?;
+        if self.dout.is_high()? {
             // Conversion not ready yet
             return Err(nb::Error::WouldBlock);
         }
 
+        // Dout falling -> clock high > 0.1 us 
+        delay.delay_us(1);
```

Comment:
Using delay makes the program yield in linux-embedded-hal (RaspberryPi). This makes the delay longer than the maximum specified 50 us. Maybe this has this been fixed in recent updates to linux-embedded-hal?

Just cycling the pin is not necessarily better, but maybe you have an Idea on how to make it work on both, Linux and HW?